Documentation/allow_legacy_mode.md

```diff
@@ -0,0 +1,101 @@
+## Allow-legacy mode
+
+etcd v2.0 uses internal API v2 protocol, which is not compatible with the v1 protocol.
```

Comment:
"uses a new internal v2 protocol"

The confusion vs v2 API may come up, but I'm not sure there's a better name. At the least "API" could be client side and "protocol" for the internal side (even though they are kind of interchangeable)